skiplang/prelude/src/skstore/Context.sk

```diff
@@ -297,6 +297,38 @@ value class Time(value: Int) uses Orderable, Show {
   }
 }
 
+class TimeStack private (private values: Array<Time>) uses Orderable {
+  static fun createInput(time: Time): this {
+    static(Array[time])
+  }
+
+  static fun create(context: mutable Context, time: Time): this {
+    context.arrowStack.maybeHead() match {
+    | None() -> static::createInput(time)
+    | Some((_arrow, timeStack)) -> timeStack.extend(time)
+    }
+  }
+
+  fun extend(time: Time): this {
+    !this.values = this.values.concat(Array[time]);
+    this
+  }
+
+  fun negate(): this {
+    !this.values = this.values.map(x -> -x);
+    this
+  }
+
+  fun compare(other: TimeStack): Order {
+    for (i in Range(0, this.values.size())) {
+      if (i >= other.values.size()) return GT();
+      cmp = this.values.compare(other.values);
```

Comment:
I don't have enough knowledge of SKStore yet to understand the rest but this is fishy: `cmp` is invariant.

Reply:
Thanks! good catch!